server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -480,4 +481,55 @@ private TranslogDeletionPolicy getTranslogDeletionPolicy(EngineConfig engineConf
         );
     }
 
+    public synchronized void finalizeReplication(SegmentInfos infos, Store.MetadataSnapshot expectedMetadata, long seqNo)
+        throws IOException {
+        assert engineConfig.isReadOnlyReplica() : "Only replicas should update Infos";
+
+        store.incRef();
+        try {
+            refreshLastCommittedSegmentInfos();
+            // clean up the local store of old segment files
+            // and validate the latest segment infos against the snapshot sent from the primary shard.
+            store.cleanupAndVerify(
+                "finalize - clean with in memory infos",
+                expectedMetadata,
+                store.getMetadata(infos),
+                store.getMetadata(lastCommittedSegmentInfos)
+            );
+        } finally {
+            store.decRef();
+        }
+        // Update the current infos reference on the Engine's reader.
+        readerManager.updateSegments(infos);
+        localCheckpointTracker.fastForwardProcessedSeqNo(seqNo);
+        readerManager.maybeRefresh();
+    }
+
+    private void refreshLastCommittedSegmentInfos() {
```

Comment:
I don't think this method is required, we already set the lastCommittedSegmentInfos inside of `updateSegments` with the passed in infos.